snuba/query/validation/validators.py

```diff
@@ -255,3 +258,58 @@ def validate(self, query: Query, alias: Optional[str] = None) -> None:
             raise InvalidQueryException(
                 f"granularity must be multiple of {self.minimum}"
             )
+
+
+class TagConditionValidator(QueryValidator):
+    """
+    Sometimes a query will have a condition that compares a tag value to a
+    non-string literal. This will always fail (tags are always strings) so
+    catch this specific case and reject it.
+    """
+
+    def __init__(self) -> None:
+        self.condition_matcher = build_match(
+            subscriptable="tags",
```

Comment:
I think the `subscriptable` parameter should be an argument in the `__init__`

Reply:
I thought about that but this is a pretty specific use case so I decided to keep it limited to tags for now.